src/aktualizr_secondary/secondary_tcp_server.cc

```diff
@@ -164,6 +175,12 @@ void SecondaryTcpServer::HandleOneConnection(int socket) {
     } else {
       LOG_DEBUG << "Not sending a response to message " << msg->present();
     }
+
```

Comment:
I don't think that there is much reason in calling `stop` here as it's mainly intended for the server stooping from another thread. Perhaps just returning a bool/int from HandleOneConnection() and handling in the while loop would be slightly better (you could set keep_running_ = false; or just `break` directly there).
Also, the name `may_reboot` a bit confusing, it's rather `want reboot` or `require reboot`, just minor :)